middleware-moshi/src/test/java/com/nytimes/android/external/store3/middleware/moshi/MoshiSourceParserTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.nytimes.android.external.store3.middleware.moshi
+
+import com.nytimes.android.external.store3.base.Fetcher
+import com.nytimes.android.external.store3.base.Persister
+import com.nytimes.android.external.store3.base.impl.BarCode
+import com.nytimes.android.external.store3.base.impl.ParsingStoreBuilder
+import io.reactivex.Maybe
+import io.reactivex.Single
+import okio.BufferedSource
+import okio.Okio
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.mockito.Mock
+import org.mockito.Mockito.*
+import org.mockito.MockitoAnnotations
+import java.io.ByteArrayInputStream
+import java.nio.charset.Charset
+
+class MoshiSourceParserTest {
+    @Rule
+    @JvmField
+    var expectedException = ExpectedException.none()
+    @Mock
+    internal var fetcher: Fetcher<BufferedSource, BarCode>? = null
+    @Mock
+    internal var persister: Persister<BufferedSource, BarCode>? = null
```

Comment:
can we make the mocks `lateinit` here and in other test classes? that way we won't need null assertions whenever they are used

Reply:
done!